src/MediatR/AsyncRequestHandler.cs

```diff
@@ -11,7 +11,7 @@ public abstract class AsyncRequestHandler<TMessage> : IAsyncRequestHandler<TMess
     {
         public async Task<Unit> Handle(TMessage message)
         {
-            await HandleCore(message);
+            await HandleCore(message).ConfigureAwait(false);
```

Comment:
Since MediatR is a library that clients will call and consume, are you not better off using ConfgureAwait(true) so the caller can decide the what the continuation context should be?